openedx/features/colaraz_features/auth_pipeline.py

```diff
@@ -1,12 +1,26 @@
 """
 Authenitcation and Social Auth Pipeline methods for Colaraz's customizations
 """
+from openedx.features.colaraz_features.models import ColarazUserProfile
+
 
 def store_id_token(request, response, user=None, *args, **kwargs):
     """
     This method is used in SOCIAL_AUTH_PIPELINE. It stores 'id_token' from the User's
     data sent by Auth Provider to request's 'session' object.
     """
-    
     if user and response.has_key('id_token'):
         request.session['id_token'] = response['id_token']
+
+
+def update_colaraz_profile(request, response, user=None, *args, **kargs):
+    """
+    Updates ColarazUserProfile using data sent from Colaraz Auth Provider
+    """
+    if user:
+        instance, _ = ColarazUserProfile.objects.update_or_create(user=user, defaults={
+            'site_identifier': str(response.get('companyInfo', {}).get('url', '')).lower(),
+            'level_percentage': response.get('levelPercentage', 0),
+            'level_text': response.get('levelText', 'Beginner'),
```

Comment:
Are there any constants that we can use for these levels ?

Reply:
The data sent by auth provider is usually in `unicode`, so have to convert that into `str` to use `lower()` method on it.
Made constants.